rcl/src/rcl/logging_rosout.c

```diff
@@ -236,10 +237,31 @@ void rcl_logging_rosout_output_handler(
 {
   rosout_map_entry_t entry;
   rcl_ret_t status = RCL_RET_OK;
+  char * name_dup = NULL;
+  rcutils_allocator_t allocator = rcutils_get_default_allocator();
   if (!__is_initialized) {
     return;
   }
   rcutils_ret_t rcutils_ret = rcutils_hash_map_get(&__logger_map, &name, &entry);
+  // if logger name not found, continue to check if its parent logger name exists
+  if (RCUTILS_RET_OK != rcutils_ret) {
+    name_dup = rcutils_strdup(name, allocator);
+    if (NULL == name_dup) {
+      RCUTILS_SAFE_FWRITE_TO_STDERR("unable to allocate memory\n");
+      return;
+    }
+    do {
+      char * name_sep = strrchr(name_dup, RCUTILS_LOGGING_SEPARATOR_STRING[0]);
```

Comment:
This is making an assumption that the `RCUTILS_LOGGING_SEPARATOR_STRING` is only a single character.  While that is true (and is unlikely to change anytime soon), it does seem weird that it is called "string", but then we are using it as a character.  Instead of using `strrchr`, can we search backwards in the string for a substr?